enhancements/authentication/automated-service-ca-rotation.md

```diff
@@ -102,29 +102,78 @@ and the functionality could be separately extracted for operator reuse.
+      - *R* = *D* - *M*
+    - Reordering to solve for *D*:
+      - *D* = *R* + *M*
+    - Since each of *R* and *M* must be greater than *I*:
+      - *D* = *R* + *M* > 2 * *I*
+    - Since *R* >= *M* should be true, simplify to *R* = *M*:
+      - *D* = 2 * *R* > 2 * *I*
+    - Substitute *I* = 12:
+      - *D* = 2 * *R* > 24
+    - Picking *R* = *M* = 13 will satisfy the relation, resulting in *D* = 26
+  - Worst-case timelines with old and new values for CA duration:
+    - Let minimum duration *M* = 6 months, total duration *D* = 12 months:
       - T+0m  - Cluster installed with new CA or existing CA is rotated (CA-1)
       - T+6m  - Automated rotation replaces CA-1 with CA-2 when CA-1 duration < 6m
       - T+12m - Cluster is upgraded and all pods are restarted
       - T+12m - CA-1 expires. If cluster was not upgraded before this
                 happens, services using the old key material may
                 break.
-    - With 14 month CA duration:
+    - Let minimum duration *M* = 13 months, total duration *D* = 26 months:
       - T+0m  - Cluster installed with new CA or existing CA is rotated (CA-1)
-      - T+8m  - Automated rotation replaces CA-1 with CA-2 when CA-1 duration < 6m
       - T+12m - Cluster is upgraded and all pods are restarted
-      - T+14m - CA-1 expires. No impact because of the restart at time of upgrade
-  - Services that do not refresh key material automatically must be restarted
-    after ca rotation.
-  - Upgrades restart all pods, and ensuring that an upgrade takes place after
-    rotation and prior to expiry of the pre-rotation ca guarantees that
-    services will always be using valid key material.
-  - If the ca duration does not exceed the expected upgrade interval, there is
-    the possibility, however slight, that an upgrade will not occur after
-    automated rotation and before expiry of the pre-rotation ca.
+      - T+13m - Automated rotation replaces CA-1 with CA-2 when CA-1 duration < 13m
+      - T+24m - Cluster is upgraded and all pods are restarted
+      - T+26m - CA-1 expires. No impact because of the restart at time of upgrade
+  - If the service CA validity duration (entire, not remaining) is less than
+    26 months, then automated rotation should be triggered and the cluster
+    administrator will need to manually restart all pods.
+    - Without this intervention, pod restart before expiry of the pre-rotation
+      CA cannot be guaranteed.
+    - Since all clusters that do not support automated rotation will have CAs
+      with a 12 month duration, rotation will be triggered at time of upgrade
+      to a release that supports automated CA rotation.
+    - The requirement to manually restart pods after a cluster is upgraded to
+      a release that supports automated CA rotation is a one-time thing. All
+      subsequent upgrades and rotations will ensure restart before expiry
+      without manual intervention.
```

Comment:
This should not include restarting control-plane pods that should be rotated by their operators

Reply:
Why would it make sense to differentiate between pods running user workloads and control plane pods of manual restart involves restarting all pods?

Reply:
They should be capable of rotating the certs for their payloads themselves, but I suppose it's safer to restart all anyway...